pyuvdata/uvbeam/uvbeam.py

```diff
@@ -607,6 +633,128 @@ def __init__(self):
 
         super(UVBeam, self).__init__()
 
+    @property
+    def _freq_params(self):
+        """List of strings giving the parameters that shaped like the freq_array."""
+        return [
+            "bandpass_array",
+            "receiver_temperature_array",
+            "loss_array",
+            "mismatch_array",
+            "freq_array",
+        ]
+
+    def _set_future_array_shapes(self):
+        """
+        Set future_array_shapes to True and adjust required parameters.
+
+        This method should not be called directly by users; instead it is called
+        by file-reading methods and `use_future_array_shapes` to indicate the
+        `future_array_shapes` is True and define expected parameter shapes.
+
+        """
+        self.future_array_shapes = True
+        self._freq_array.form = ("Nfreqs",)
+        for param_name in self._freq_params:
+            getattr(self, "_" + param_name).form = ("Nfreqs",)
+
+        self._s_parameters.form = (4, "Nfreqs")
+        self._data_array.form = ("Naxes_vec", "Nfeeds", "Nfreqs", "Naxes2", "Naxes1")
+        self._coupling_matrix.form = (
+            "Nelements",
+            "Nelements",
+            "Nfeeds",
+            "Nfeeds",
+            "Nfreqs",
+        )
```

Comment:
I can, but I'd actually have to look for a "1" not Nspws because I explicitly removed Nspws from the forms. I did that because we've never supported Nspws > 1.

Reply:
Ah. Well, that seems brittle.